test/test_torch.py

```diff
@@ -9554,6 +9554,28 @@ def test_cdist_norm_batch(self, device):
                         expected = brute_cdist(x, y, p=p)
                         self.assertTrue(torch.allclose(expected, actual))
 
+    def test_multinomial_unnormalized(self, device):
+        for dtype in [torch.float, torch.double]:
```

Comment:
It would probably be nice to check for CUDA half as well? But that test might (and I hope) exists already